rules/cloud/policies/aws/s3/enable_versioning.go

```diff
@@ -41,7 +41,8 @@ With versioning you can recover more easily from both unintended user actions an
 	},
 	func(s *state.State) (results scan.Results) {
 		for _, bucket := range s.AWS.S3.Buckets {
-			if !bucket.Versioning.Enabled.IsTrue() {
+			// Versioning check changed as Object lock can be configured on versioned buckets only
+			if !bucket.Versioning.Enabled.IsTrue() && !bucket.ObjectLockConfiguration.Enabled.IsTrue() {
```

Comment:
Sorry but I don't follow, why does this logic needs to be changed? Your comment mentions that Object lock can only be configured on versioned buckets only but this check only checks for versioning to be enabled (or otherwise) for buckets.

Reply:
![Screenshot from 2023-03-16 11-08-58](https://user-images.githubusercontent.com/18326772/225525279-d68564a4-ac24-4dc2-8af4-07b26b696575.png)
this is ticket description that needed to be implemented...as per point iii have added this condition

Reply:
Hi @VikrantCGaikwad if Object Lock is enabled, Versioning must be enabled. But if Versioning is enabled, Object Lock does not need to be enabled. This change will break our functionality.

Reply:
Hi @giorod3 @simar7 ...
if !bucket.Versioning.Enabled.IsTrue() && !bucket.ObjectLockConfiguration.Enabled.IsTrue() {
If this condition is true then we are seeting only Versioning not Object Lock Configuration...
**But if Versioning is enabled, Object Lock does not need to be enabled**  we are not setting this anywhere